src/slack_interaction_handler.ts

```diff
@@ -1193,6 +1196,124 @@ export async function receiveResetDemo({
+}): Promise<void> {
+  logger.info(`Entering SLACKINTERACTIONHANDLER.receiveRouteToJourney`);
+  let slackView;
+
+  try {
+    const redisUserInfoKey = `${userId}:${twilioPhoneNumber}`;
+    const userInfo = (await RedisApiUtil.getHash(
+      redisClient,
+      redisUserInfoKey
+    )) as UserInfo;
+
+    if (!userInfo) {
+      modalPrivateMetadata.success = false;
+      modalPrivateMetadata.failureReason = 'no_user_info';
+      await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+      throw new Error(
+        `SLACKINTERACTIONHANDLER.receiveRouteToJourney: Interaction received for voter who has redisData but not userInfo: active redisData key is ${payload.channel.id}:${payload.message.ts}, userInfo key is ${redisUserInfoKey}.`
+      );
+    }
+
+    if (!(payload.channel.id === userInfo.activeChannelId)) {
+      modalPrivateMetadata.success = false;
+      modalPrivateMetadata.failureReason = 'non_active_thread';
+      await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+      logger.info(
+        `SLACKINTERACTIONHANDLER.receiveRouteToJourney: Volunteer issued route to journey command from #${payload.channel.id} but voter active channel is ${userInfo.activeChannelId}.`
+      );
+      slackView = SlackBlockUtil.getErrorSlackView(
+        'demo_reset_error_not_active_thread',
+        `This voter is no longer active in this thread. Please reach out to the folks at *#${userInfo.activeChannelName}*.`
+      );
+    } else {
+      const activeSlackChannelName = userInfo.activeChannelName;
+      const activeStateName = LoadBalancer.convertSlackChannelNameToStateOrRegionName(
+        activeSlackChannelName
+      );
+
+      if (!activeStateName) {
+        slackView = SlackBlockUtil.getErrorSlackView(
+          'route_to_journey_error_no_state_name_parsed',
+          `Failed to parse state name from channel: *#${activeSlackChannelName}*. Please contact an admin.`
+        );
+      } else {
+        const selectedSlackChannelName = await LoadBalancer.selectSlackChannel(
+          redisClient,
+          // We use PUSH to store open journey pods (PULL = frontline).
+          'PUSH',
+          activeStateName,
+          userInfo.isDemo
+        );
+
+        // Either a stateName wasn't valid or Redis didn't provide open pods for the given stateName.
+        if (
+          !selectedSlackChannelName ||
+          ['demo-national-0', 'national-0'].includes(selectedSlackChannelName)
+        ) {
+          slackView = SlackBlockUtil.getErrorSlackView(
+            'route_to_journey_error_no_pods_found',
+            `No journey pods found for U.S. state name: *${activeStateName}*. Please contact an admin.`
+          );
+        } else if (
+          activeSlackChannelName === selectedSlackChannelName
+        ) {
+          slackView = SlackBlockUtil.getErrorSlackView(
+            'route_to_journey_error_routing_to_same_channel',
+            `The voter is already in an open journey pod for this U.S. state.`
+          );
```

Comment:
The condition and error message don't completely match. IIUC this basically means that if you have multiple possible target channels for the state, and the voter is already in one of them, then sometimes we'll randomly pick the same one and get this error message (which will make sense in that context) or we'll randomly pick a different journey pod and the voter will get routed there.  If there is a way to identify whether the current channel is a journey channel that would be better.  If not, this should probably go on a list of things to improve later!

Reply:
Ya you're describing that accurately. This is something worth following up on -- unfortunately I didn't see an easy way to do this without adding complexity to the LoadBalancer or a second Redis fetch for available pods. I think for the Georgia election we can just keep this in mind (this hopefully shouldn't happen often) -- and at least the case of routing to the same channel will be caught and stopped.

I created an issue so we can follow-up: https://github.com/tomerovadia/VoterHelpLine/issues/217

If everything else looks good do you mind merging this once #214 Session topics is in?